modules/@apostrophecms/page/index.js

```diff
@@ -988,6 +988,15 @@ database.`);
           page.path = self.apos.util.addSlashIfNeeded(parent.path) + page.aposDocId;
           page.level = parent.level + 1;
           await self.apos.doc.insert(req, page, options);
+          // Prevent a published page from being inserted as a child of a draft page.
```

Comment:
This is this check not happening until after the insert? We can tell if this is going to be a published page or not right away based on its mode. Then we would not have to publish and unpublish it, momentarily breaking the rule.